src/backend/optimizer/prep/prepunion.c

```diff
@@ -1808,6 +1808,40 @@ adjust_appendrel_attrs(PlannerInfo *root, Node *node, AppendRelInfo *appinfo)
 	return result;
 }
 
+static bool
+nested_subplans_mutator(Node *node, adjust_appendrel_attrs_context *context) {
+	if (node == NULL) {
+		return false;
+	}
+
+	if (IsA(node, SubPlan)) {
+		SubPlan *sp = (SubPlan*) node;
+
+		if (!sp->is_initplan) {
+			PlannerInfo *root = context->root;
+			Plan *newsubplan = (Plan *) copyObject(planner_subplan_get_plan(root, sp));
+			PlannerInfo *newsubroot = makeNode(PlannerInfo);
+
+			memcpy(newsubroot, planner_subplan_get_root(root, sp), sizeof(PlannerInfo));
+
+			plan_tree_walker(newsubplan, nested_subplans_mutator, context);
```

Comment:
`plan_tree_walker` expects that context is a pointer to `plan_tree_base_prefix`-compatible struct ([here](https://github.com/arenadata/gpdb/blob/ADBDEV-2501/src/backend/optimizer/util/walkers.c#L439)). Can you explain why do you use `adjust_appendrel_attrs_context` instead?

Reply:
I've added a new struct - nested_subplans_context, that's a derivative of plan_tree_base_prefix

Reply:
I think that defining struct name in both typedef and tag namespaces is more common pattern for gpdb code base.

Reply:
Can you explain what you mean about tag namespace? Is that a value for a `type` field of a `Node`?

Reply:
`typedef struct TagNamespaceName {} TypedefNamespaceName`